app/scripts/directives/processTemplate.js

```diff
@@ -226,7 +230,13 @@
         };
         ctrl.template.labels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries(ctrl.labels));
 
-        DataService.create("processedtemplates", null, ctrl.template, context).then(
+        // NOTE:
```

Comment:
This is worth some discussion, I'm guessing there may be some potential problems with this approach, however, it fixes the mismatch problem and appears to be what the CLI is doing.  However:

```JavaScript 
// given 
preferredVersion = {group: 'template.openshift.io',      version: 'v1',      resource: 'processedtemplates' };
// if template is:
apiVersion = 'v1';                                      // update it to template.openshift.io/v1 
apiVersion = 'template.openshift.io/v1'  // updating, but its the same, so no foul 
apiVersion = 'template.openshift.io/v2' //  hmm... our preferredVersion is v1 so
apiVersion = 'template.something.else/v5' // what to do with unexpected? should there ever be?
```
So this fixes the problem at the moment, is it the correct fix long term?